### DISCUSSION TITLE
eslint --fix Command failed with exit code 2.

### DISCUSSION THREAD

**Discussion by suming1016** [eslint/eslint#18692] [Q&A]
运行环境：vue3+ts+eslint9.x
我的配置 eslint.config.mjs
`import globals from "globals";
import pluginJs from "@eslint/js";
import tseslint from "typescript-eslint";
import pluginVue from "eslint-plugin-vue";
import eslintConfigPrettier from "eslint-config-prettier";
import eslintPluginImport from "eslint-plugin-import";
import pluginPrettierRecommendedConfigs from "eslint-plugin-prettier/recommended"
import parserVue from "vue-eslint-parser"
export default [
  {
    files: ["**/*.{js,mjs,cjs,ts,vue}"],
    ignores: ["**/*.config.js", "!**/eslint.config.js", "node_modules/*", "dist/*", "build/**/*"]
  },
  {
    languageOptions: {
      globals: { ...globals.browser, ...globals.node },
      parser: parserVue,
    }
  },
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  ...pluginVue.configs["flat/essential"],
  eslintConfigPrettier, eslintPluginImport, pluginPrettierRecommendedConfigs,
  {
    files: ["**/*.vue"], languageOptions: { parserOptions: { parser: tseslint.parser } }
  }, {
    rules: {
      "no-unused-vars": "error",
      "prettier/prettier": "error",
      "import/no-unresolved": "off",
      /* 对后缀的检测，否则 import 一个ts文件也会报错，需要手动添加".ts", 增加了下面的配置后就不用了*/
      "import/extensions": [
        "error",
        "ignorePackages",
        { js: "never", jsx: "never", ts: "never", tsx: "never" },
      ],
      /* 只允许1个默认导出，关闭，否则不能随意export xxx*/
      "import/prefer-default-export": ["off"],
      "no-console": ["off"],
      "vue/multi-word-component-names": "off",
      /* 避免 `eslint` 对于 `typescript` 函数重载的误报*/
      "no-redeclare": "off",
    }
  }
];
`
报错：`> eslint --fix

Oops! Something went wrong! :(

ESLint: 9.7.0

Configuration for rule "no-unresolved" is invalid. Each rule must have a severity ("off", 0, "warn", 1, "error", or 2) and may be followed by additional options for the rule.

You passed '{
        "meta": {
            "type": "problem",
            "docs": {
                "category": "Static analysis",
                "description": "Ensure imports point to a file/module that can be resolved.",
                "url": "https://github.com/import-js/eslint-plugin-import/blob/v2.29.1/docs/rules/no-unresolved.md"
            },
            "schema": [
                {
                    "type": "object",
                    "properties": {
                        "commonjs": {
                            "type": "boolean"
                        },
                        "amd": {
                            "type": "boolean"
                        },
                        "esmodule": {
                            "type": "boolean"
                        },
                        "ignore": {
                            "type": "array",
                            "minItems": 1,
                            "items": {
                                "type": "string"
                            },
                            "uniqueItems": true
                        },
                        "caseSensitive": {
                            "type": "boolean",
                            "default": true
                        },
                        "caseSensitiveStrict": {
                            "type": "boolean",
                            "default": false
                        }
                    },
                    "additionalProperties": false
                }
            ]
        }
    }', which doesn't contain a valid severity.

If you're attempting to configure rule options, perhaps you meant:

    "no-unresolved": ["error", {
            "meta": {
                "type": "problem",
                "docs": {
                    "category": "Static analysis",
                    "description": "Ensure imports point to a file/module that can be resolved.",
                    "url": "https://github.com/import-js/eslint-plugin-import/blob/v2.29.1/docs/rules/no-unresolved.md"
                },
                "schema": [
                    {
                        "type": "object",
                        "properties": {
                            "commonjs": {
                                "type": "boolean"
                            },
                            "amd": {
                                "type": "boolean"
                            },
                            "esmodule": {
                                "type": "boolean"
                            },
                            "ignore": {
                                "type": "array",
                                "minItems": 1,
                                "items": {
                                    "type": "string"
                                },
                                "uniqueItems": true
                            },
                            "caseSensitive": {
                                "type": "boolean",
                                "default": true
                            },
                            "caseSensitiveStrict": {
                                "type": "boolean",
                                "default": false
                            }
                        },
                        "additionalProperties": false
                    }
                ]
            }
        }]

See https://eslint.org/docs/latest/use/configure/rules#using-configuration-files for configuring rules.`

**Comment by aladdin-add** [comment]
@suming1016 can you provide a repro? we are not able to help without detailed info.

**Comment by Selithrarion** [comment]
if you import
import pluginImport from 'eslint-plugin-import'

use
pluginImport.flatConfigs.recommended,

instead of
pluginImport,

it looks like this

import pluginImport from 'eslint-plugin-import'
export default defineConfigWithVueTs(
...
pluginImport.flatConfigs.recommended,

that helped me
